### PR TITLE
Fix PR #3598 : Auto init non-git repository (part 2)

### PR DESCRIPTION
**Pull request by KaKi87**
Fixes #3598

### CODE REVIEW

**Comment by mittalyashu** [comment]
I tested the fix locally on my machine.

Seems like everything is working fine, except the button status is not getting updated when `autoInit` is enabled.

https://youtu.be/dkjWGIyi1dw, please see this video.

**Comment by KaKi87** [comment]
I see. However, this is out of this PR's scope.
Could you please merge this one and close #3598 then I'll be working on a fix for this button ?
Thanks

**Comment by mittalyashu** [comment]
That button issue is surely a part of this PR, could you please add a fix for the same.

**Comment by KaKi87** [comment]
The bug you're describing predates this PR.

**Comment by mittalyashu** [comment]
I understand what you mean, that this PR title describes to add feature to auto-initialise the git repository, in addition to that initialising the git repository from button is also a part of it.

File: src/renderer/mixins/addRepository.js [hunk 1]
```diff
@@ -66,8 +67,16 @@ export default {
...
+				let repository;
+				try {
+					repository = await nodegit.Repository.open(path);
+				} catch (_) {}
+				let autoInitFeature = this.$store.getters["settings/getExperimental"]
+					.autoInit;
+				if (!repository && autoInitFeature) {
+					gitInit({ path });
+				}
 				this.newRepository.isGitRepo = !!repository;
```
Comment (mittalyashu) [inline src/renderer/mixins/addRepository.js]:
```suggestion
				this.newRepository.isGitRepo = true;
```

Comment (mittalyashu) [reply src/renderer/mixins/addRepository.js]: I tried few ways to get the `Repository {}` object, seems like it not working. So for now we go with hard-coded true value.

Not the best option, I know, but I will look into improving the gitInit API further.

File: src/renderer/modal/newRepository.vue [hunk 2]
```diff
@@ -102,13 +101,8 @@ export default {
...
 				fs.mkdirSync(repositoryPath);
 			}
-			this.gitInit(repositoryPath);
 			this.closeModal("NewRepository");
-			this.localRepository(repositoryPath);
-		},
-		async gitInit(path) {
-			await gitInit({ path });
-			this.addRepositoryToDatabase(path);
+			this.addRepositoryToDatabase(repositoryPath);
```
Comment (mittalyashu) [inline src/renderer/modal/newRepository.vue]:
```suggestion
			this.addRepositoryToDatabase(repositoryPath);
```

Can you move this before closing the modal?

File: src/renderer/components/repositoryListView/repositoryItem.vue [hunk 3]
```diff
@@ -106,11 +106,13 @@ export default {
...
+				`
 				UPDATE gitRepository
 				SET isGit = TRUE
 				WHERE repositoryId = $repositoryId
-			`, { $repositoryId: this.repositoryId },
+			`,
+				{ $repositoryId: this.repositoryId },
 				(err, data) => {
 					if (err) console.log(err);
 					else this.isGit = 1;
```
Comment (mittalyashu) [inline src/renderer/components/repositoryListView/repositoryItem.vue]:
```suggestion
					else this.queryAllRepository();
```

Also, don't forget to import the required mixin for the same.

**Comment by KaKi87** [comment]
> @KaKi87 Can you please look into these changes, after then I think we're good with merging it.

Again, your changes request would only be justified if my PR was creating regressions, which is not the case.

Please let's work step-by-step.

**Comment by KaKi87** [comment]
Finally.
I'll start working on the several issues you talked about.

**Pull request merged**